odoo/api/__init__.py

```diff
@@ -0,0 +1,23 @@
+# ruff: noqa: F401
+# Exports features of the ORM to developers.
+# This is a `__init__.py` file to avoid merge conflicts on `odoo/api.py`.
+import typing
+
+from odoo._orm.identifier import NewId
+from odoo._orm.api import (
+    Environment,
+    autovacuum,
+    call_kw,
+    constrains,
+    depends,
+    depends_context,
+    model,
+    model_create_multi,
+    onchange,
+    ondelete,
+    readonly,
+    returns,
+)
+
+if typing.TYPE_CHECKING:
+    from odoo._orm.types import BaseModel, ContextType, DomainType, IdType, NewId, Self, ValuesType
```

Comment:
Some of those types were always available.  Shouldn't we remain backward compatible?
```suggestion
if typing.TYPE_CHECKING:
    from odoo._orm.types import BaseModel, IdType, NewId, Self
from odoo._orm.types import ContextType, DomainType, ValuesType
```

Reply:
No, because these are introduced only for type-checking, importing `odoo._orm.types` without typing context will lead to cycles in dependencies.

Reply:
Exposed as before.